smpp/server.go

```diff
@@ -127,6 +125,21 @@ func NewUnstartedServer(user, password string, listener net.Listener) Server {
 	return s
 }
 
+// NewTLDUnstartedServer creates a new SSL Server with default settings, and
+// does not start it. Callers are supposed to call Start and Close later.
+func NewTLSUnstartedServer(user, password string, listener net.Listener, tlsConf *tls.Config) Server {
+	s := &server{
+		User:   user,
+		Passwd: password,
+		TLS:    tlsConf,
+		m:      map[pdu.ID]RequestHandlerFunc{},
+		a:      map[pdu.ID]AuthRequestHandlerFunc{},
+		s:      map[string]Session{},
+		l:      listener,
+	}
+	return s
+}
+
```

Comment:
thanks, I think I found a better way to do this without actually modifying go-smpp, will ping you to have another look at the esme PR, hopefully we could scrap this PR.